### PR TITLE
Add dependabot PR merger script tool

### PR DESCRIPTION
**Pull request by rapphil**
**Description:** Use https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/.github/workflows/scripts/dependabot-pr.sh as reference to create a script tool to merge dependabot prs.

**Testing:** Tested locally and created the following PR: https://github.com/aws-observability/aws-otel-collector/pull/1871

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
You still need to add a target in `Makefile` to ensure the target in `Makefile.common` gets triggered. See golint for an example
```
.PHONY: golint
golint: lint-static-check
	@$(MAKE) for-all-target TARGET="lint"
```

**Comment by rapphil** [comment]
> 



> You still need to add a target in `Makefile` to ensure the target in `Makefile.common` gets triggered. See golint for an example
> 
> ```
> .PHONY: golint
> golint: lint-static-check
> 	@$(MAKE) for-all-target TARGET="lint"
> ```

I can change, but I was able to use the target from the script anyways. 
https://github.com/aws-observability/aws-otel-collector/pull/1872/files#diff-96e901e1ff7755937b39e1e13ad213df1418d528e975004f62c06221eb6cfc39R58

**Comment by bryan-aguilar** [comment]
I think we should move that into the makefile. There are reasons that an operator would want to run go mod tidy outside of updating the dependabot PRs. Moving it into the makefile would make it easier for them.

**Comment by rapphil** [comment]
> I think we should move that into the makefile. There are reasons that an operator would want to run go mod tidy outside of updating the dependabot PRs. Moving it into the makefile would make it easier for them.

done! @bryan-aguilar

File: Makefile [hunk 1]
```diff
@@ -181,6 +181,10 @@ lint-static-check:
 golint: lint-static-check
 	@$(MAKE) for-all-target TARGET="lint"
 
+.PHONY: gomod-tidy
+gomod-tidy:
```
Comment (rapphil) [inline Makefile:185]: this target name follows the same convention of merging `go` with the command name

**Review by bryan-aguilar — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**